// -----------------------------------------
// 好友联系人相关
// -----------------------------------------

//好友基本信息
message DFriendInfo {
	optional int64 humanId = 1;		//人物Id
	optional string humanName = 2;	//名字
	optional int32 level = 3;		//等级

	optional bool isOnline = 5;		//是否在线
	optional string intorduction = 6;	//个性签名
	optional int64 timeLogout 	 = 7;	//上次登出时间，如果isOnline是true，忽略这个字段
	optional string backName 	 = 8;	//备注名称，可能为空
	optional int64 identifier 	 = 9;	//标识ID
	optional int32 portrait 	 = 10;	//角色头像
	optional int32 portraitFrame = 11;	//角色头像框 
	optional int64 upTime	 	 = 12;	//置顶时间，0或者-1，表示没有置顶或取消置顶
	optional int32 assistThiefSn 		= 13;	//协助怪盗sn
	optional int32 assistThiefGrade		= 14;	//协助怪盗品质
	optional int32 assistThiefLevel		= 15;	//协助怪盗等级
}

//搜索好友
message CSSearchFriend {
//    option (msgid) = 1901;
	optional string searchName = 1;		//根据名字搜索玩家
}

//推荐好友返回
message SCRecommendFriend  {
//    option (msgid) = 1902;
	repeated DFriendInfo recomlist 	= 1; //推荐好友返回
	optional int32 errorCode 		= 2; //0为正常，其他都是错误
}

//获取黑名单列表
message CSGetBlacklist {
//    option (msgid) = 1903;
}

//获取黑名单列表返回
message SCGetBlacklist {
//    option (msgid) = 1904;
    repeated DFriendInfo list = 1; //玩家列表
}

//增加黑名单
message CSAddBlacklist {
//	option (msgid) = 1905;
	optional int64 humanId = 1;		    //玩家Id
}

//删除黑名单
message CSRemoveBlacklist {
//	option (msgid) = 1906;
	repeated int64 humanIds = 1;		    //玩家Id
}

//获取好友列表
message CSGetFriendList {
//    option (msgid) = 1907;
}

//获取好友列表返回
message SCGetFriendList {
//    option (msgid) = 1908;
    repeated DFriendInfo friendlist = 1; //玩家列表
}

//获取申请列表
message CSGetRequestList {
//    option (msgid) = 1909;
}

message DRequestTime {
	optional int64 humanId 		= 1;	//人物Id
	optional int64 timeRequest 	= 2;	//申请时间
}

//获取申请列表返回
message SCGetRequestList {
//    option (msgid) = 1910;
    repeated DFriendInfo friendlist = 1; //申请玩家列表
	repeated DRequestTime friendReqTime = 2; //申请时间，用于计算有效期
}

//申请好友
message CSRequestFriend {
//    option (msgid) = 1911;
	repeated int64 beReqId = 1;		    //被邀请的人Id
}

//申请好友返回
message SCRequestFriend {
//    option (msgid) = 1912;
	optional bool isSuccess = 1;	    // 是否申请成功
}

//接受好友
message CSAcceptFriend {
//    option (msgid) = 1913;
	repeated int64 reqId = 1;		    //要接受的人的ID列表
}

//拒绝好友
message CSRefuseFriend {
//    option (msgid) = 1914;
	repeated int64 refuseId = 1;		    //要拒绝的人的ID列表
}

//删除好友
message CSRemoveFriend {
//	option (msgid) = 1915;
	repeated int64 humanIds = 1;	//玩家Id
}

//推荐好友
message CSRecommendFriend {
//    option (msgid) = 1916;
}

//给好友备注
message CSNameFriend {
//    option (msgid) = 1917;
	optional int64 friendId = 1; // 好友ID
	optional string name 	= 2; // 备注名称
}

//给好友备注
message SCNameFriend {
//    option (msgid) = 1918;
	optional int64 friendId = 1; // 好友ID
	optional string name 	= 2; // 备注名称
	optional int32 errorCode= 3; // 错误码，0是正常
}

//接受好友
message SCAcceptFriend {
//    option (msgid) = 1919;
	repeated int32 errorCode = 1;		    //多个错误
}

//置顶好友
message CSUpFriend {
//    option (msgid) = 1920;
	optional int64 friendId = 1; // 好友ID
}

//置顶好友
message SCUpFriend {
//    option (msgid) = 1921;
	optional int64 friendId = 1; // 好友ID
	optional int32 error 	= 2; // 0正确
	optional int64 upTime	= 3; // 时间
}

//取消置顶好友
message CSCancelUpFriend {
//    option (msgid) = 1922;
	optional int64 friendId = 1; // 好友ID
}

//取消置顶好友
message SCCancelUpFriend {
//    option (msgid) = 1923;
	optional int64 friendId = 1; // 好友ID
	optional int32 error 	= 2; // 0正确
}

//查看协助怪盗属性
message CSReqAssistInfo {
//    option (msgid) = 1924;
	optional int64 humanId = 1; // 玩家ID
}

// 属性信息
message DUnitProp {
	// GENERATE BEGIN
	// 注意：下面代码是由工具：org.jow.scenesrv.moduleUnit.attr.gen.GenAttrProto生成出来的，请勿手动修改
	optional float LL		= 1;
	optional float LL_base		= 2;
	optional float LL_add		= 3;
	optional float LL_ratio		= 4;
	optional float LL_sratio		= 5;
	optional float ML		= 6;
	optional float ML_base		= 7;
	optional float ML_add		= 8;
	optional float ML_ratio		= 9;
	optional float ML_sratio		= 10;
	optional float NL		= 11;
	optional float NL_base		= 12;
	optional float NL_add		= 13;
	optional float NL_ratio		= 14;
	optional float NL_sratio		= 15;
	optional float YQ		= 16;
	optional float YQ_base		= 17;
	optional float YQ_add		= 18;
	optional float YQ_ratio		= 19;
	optional float YQ_sratio		= 20;
	optional float HPcur		= 21;
	optional float HPcur_base		= 22;
	optional float HPcur_add		= 23;
	optional float HPcur_ratio		= 24;
	optional float HPcur_sratio		= 25;
	optional float SPcur		= 26;
	optional float SPcur_base		= 27;
	optional float SPcur_add		= 28;
	optional float SPcur_ratio		= 29;
	optional float SPcur_sratio		= 30;
	optional float TLcur		= 31;
	optional float TLcur_base		= 32;
	optional float TLcur_add		= 33;
	optional float TLcur_ratio		= 34;
	optional float TLcur_sratio		= 35;
	optional float HP		= 36;
	optional float HP_base		= 37;
	optional float HP_add		= 38;
	optional float HP_ratio		= 39;
	optional float HP_sratio		= 40;
	optional float SP		= 41;
	optional float SP_base		= 42;
	optional float SP_add		= 43;
	optional float SP_ratio		= 44;
	optional float SP_sratio		= 45;
	optional float GJ		= 46;
	optional float GJ_base		= 47;
	optional float GJ_add		= 48;
	optional float GJ_ratio		= 49;
	optional float GJ_sratio		= 50;
	optional float FY		= 51;
	optional float FY_base		= 52;
	optional float FY_add		= 53;
	optional float FY_ratio		= 54;
	optional float FY_sratio		= 55;
	optional float BJ		= 56;
	optional float BJ_base		= 57;
	optional float BJ_add		= 58;
	optional float BJ_ratio		= 59;
	optional float BJ_sratio		= 60;
	optional float BS		= 61;
	optional float BS_base		= 62;
	optional float BS_add		= 63;
	optional float BS_ratio		= 64;
	optional float BS_sratio		= 65;
	optional float XGMZ		= 66;
	optional float XGMZ_base		= 67;
	optional float XGMZ_add		= 68;
	optional float XGMZ_ratio		= 69;
	optional float XGMZ_sratio		= 70;
	optional float XGDK		= 71;
	optional float XGDK_base		= 72;
	optional float XGDK_add		= 73;
	optional float XGDK_ratio		= 74;
	optional float XGDK_sratio		= 75;
	optional float TL		= 76;
	optional float TL_base		= 77;
	optional float TL_add		= 78;
	optional float TL_ratio		= 79;
	optional float TL_sratio		= 80;
	optional float MZ		= 81;
	optional float MZ_base		= 82;
	optional float MZ_add		= 83;
	optional float MZ_ratio		= 84;
	optional float MZ_sratio		= 85;
	optional float ZCSH		= 86;
	optional float ZCSH_base		= 87;
	optional float ZCSH_add		= 88;
	optional float ZCSH_ratio		= 89;
	optional float ZCSH_sratio		= 90;
	optional float SDSH		= 91;
	optional float SDSH_base		= 92;
	optional float SDSH_add		= 93;
	optional float SDSH_ratio		= 94;
	optional float SDSH_sratio		= 95;
	optional float ZCWLSH		= 96;
	optional float ZCWLSH_base		= 97;
	optional float ZCWLSH_add		= 98;
	optional float ZCWLSH_ratio		= 99;
	optional float ZCWLSH_sratio		= 100;
	optional float ZCSJSH		= 101;
	optional float ZCSJSH_base		= 102;
	optional float ZCSJSH_add		= 103;
	optional float ZCSJSH_ratio		= 104;
	optional float ZCSJSH_sratio		= 105;
	optional float ZCHYSH		= 106;
	optional float ZCHYSH_base		= 107;
	optional float ZCHYSH_add		= 108;
	optional float ZCHYSH_ratio		= 109;
	optional float ZCHYSH_sratio		= 110;
	optional float ZCBDSH		= 111;
	optional float ZCBDSH_base		= 112;
	optional float ZCBDSH_add		= 113;
	optional float ZCBDSH_ratio		= 114;
	optional float ZCBDSH_sratio		= 115;
	optional float ZCDJSH		= 116;
	optional float ZCDJSH_base		= 117;
	optional float ZCDJSH_add		= 118;
	optional float ZCDJSH_ratio		= 119;
	optional float ZCDJSH_sratio		= 120;
	optional float ZCJFSH		= 121;
	optional float ZCJFSH_base		= 122;
	optional float ZCJFSH_add		= 123;
	optional float ZCJFSH_ratio		= 124;
	optional float ZCJFSH_sratio		= 125;
	optional float ZCNDSH		= 126;
	optional float ZCNDSH_base		= 127;
	optional float ZCNDSH_add		= 128;
	optional float ZCNDSH_ratio		= 129;
	optional float ZCNDSH_sratio		= 130;
	optional float ZCHRSH		= 131;
	optional float ZCHRSH_base		= 132;
	optional float ZCHRSH_add		= 133;
	optional float ZCHRSH_ratio		= 134;
	optional float ZCHRSH_sratio		= 135;
	optional float ZCZFSH		= 136;
	optional float ZCZFSH_base		= 137;
	optional float ZCZFSH_add		= 138;
	optional float ZCZFSH_ratio		= 139;
	optional float ZCZFSH_sratio		= 140;
	optional float ZCZZSH		= 141;
	optional float ZCZZSH_base		= 142;
	optional float ZCZZSH_add		= 143;
	optional float ZCZZSH_ratio		= 144;
	optional float ZCZZSH_sratio		= 145;
	optional float ZCWNSH		= 146;
	optional float ZCWNSH_base		= 147;
	optional float ZCWNSH_add		= 148;
	optional float ZCWNSH_ratio		= 149;
	optional float ZCWNSH_sratio		= 150;
	optional float SDWLSH		= 151;
	optional float SDWLSH_base		= 152;
	optional float SDWLSH_add		= 153;
	optional float SDWLSH_ratio		= 154;
	optional float SDWLSH_sratio		= 155;
	optional float SDSJSH		= 156;
	optional float SDSJSH_base		= 157;
	optional float SDSJSH_add		= 158;
	optional float SDSJSH_ratio		= 159;
	optional float SDSJSH_sratio		= 160;
	optional float SDHYSH		= 161;
	optional float SDHYSH_base		= 162;
	optional float SDHYSH_add		= 163;
	optional float SDHYSH_ratio		= 164;
	optional float SDHYSH_sratio		= 165;
	optional float SDBDSH		= 166;
	optional float SDBDSH_base		= 167;
	optional float SDBDSH_add		= 168;
	optional float SDBDSH_ratio		= 169;
	optional float SDBDSH_sratio		= 170;
	optional float SDDJSH		= 171;
	optional float SDDJSH_base		= 172;
	optional float SDDJSH_add		= 173;
	optional float SDDJSH_ratio		= 174;
	optional float SDDJSH_sratio		= 175;
	optional float SDJFSH		= 176;
	optional float SDJFSH_base		= 177;
	optional float SDJFSH_add		= 178;
	optional float SDJFSH_ratio		= 179;
	optional float SDJFSH_sratio		= 180;
	optional float SDNDSH		= 181;
	optional float SDNDSH_base		= 182;
	optional float SDNDSH_add		= 183;
	optional float SDNDSH_ratio		= 184;
	optional float SDNDSH_sratio		= 185;
	optional float SDHRSH		= 186;
	optional float SDHRSH_base		= 187;
	optional float SDHRSH_add		= 188;
	optional float SDHRSH_ratio		= 189;
	optional float SDHRSH_sratio		= 190;
	optional float SDZFSH		= 191;
	optional float SDZFSH_base		= 192;
	optional float SDZFSH_add		= 193;
	optional float SDZFSH_ratio		= 194;
	optional float SDZFSH_sratio		= 195;
	optional float SDZZSH		= 196;
	optional float SDZZSH_base		= 197;
	optional float SDZZSH_add		= 198;
	optional float SDZZSH_ratio		= 199;
	optional float SDZZSH_sratio		= 200;
	optional float SDWNSH		= 201;
	optional float SDWNSH_base		= 202;
	optional float SDWNSH_add		= 203;
	optional float SDWNSH_ratio		= 204;
	optional float SDWNSH_sratio		= 205;
	optional float SDBJ		= 206;
	optional float SDBJ_base		= 207;
	optional float SDBJ_add		= 208;
	optional float SDBJ_ratio		= 209;
	optional float SDBJ_sratio		= 210;
	optional float BJKX		= 211;
	optional float BJKX_base		= 212;
	optional float BJKX_add		= 213;
	optional float BJKX_ratio		= 214;
	optional float BJKX_sratio		= 215;
	optional float ZCZL		= 216;
	optional float ZCZL_base		= 217;
	optional float ZCZL_add		= 218;
	optional float ZCZL_ratio		= 219;
	optional float ZCZL_sratio		= 220;
	optional float SDZL		= 221;
	optional float SDZL_base		= 222;
	optional float SDZL_add		= 223;
	optional float SDZL_ratio		= 224;
	optional float SDZL_sratio		= 225;
	optional float ZCHD		= 226;
	optional float ZCHD_base		= 227;
	optional float ZCHD_add		= 228;
	optional float ZCHD_ratio		= 229;
	optional float ZCHD_sratio		= 230;
	optional float SDHD		= 231;
	optional float SDHD_base		= 232;
	optional float SDHD_add		= 233;
	optional float SDHD_ratio		= 234;
	optional float SDHD_sratio		= 235;
	optional float SKPMI		= 236;
	optional float SKPMI_base		= 237;
	optional float SKPMI_add		= 238;
	optional float SKPMI_ratio		= 239;
	optional float SKPMI_sratio		= 240;
	optional float SKPMII		= 241;
	optional float SKPMII_base		= 242;
	optional float SKPMII_add		= 243;
	optional float SKPMII_ratio		= 244;
	optional float SKPMII_sratio		= 245;
	optional float SKPMIII		= 246;
	optional float SKPMIII_base		= 247;
	optional float SKPMIII_add		= 248;
	optional float SKPMIII_ratio		= 249;
	optional float SKPMIII_sratio		= 250;
	optional float SKPMIV		= 251;
	optional float SKPMIV_base		= 252;
	optional float SKPMIV_add		= 253;
	optional float SKPMIV_ratio		= 254;
	optional float SKPMIV_sratio		= 255;
	optional float SKPMV		= 256;
	optional float SKPMV_base		= 257;
	optional float SKPMV_add		= 258;
	optional float SKPMV_ratio		= 259;
	optional float SKPMV_sratio		= 260;
	optional float SKPMVI		= 261;
	optional float SKPMVI_base		= 262;
	optional float SKPMVI_add		= 263;
	optional float SKPMVI_ratio		= 264;
	optional float SKPMVI_sratio		= 265;
	optional float SKPMVII		= 266;
	optional float SKPMVII_base		= 267;
	optional float SKPMVII_add		= 268;
	optional float SKPMVII_ratio		= 269;
	optional float SKPMVII_sratio		= 270;
	optional float SKPMVIII		= 271;
	optional float SKPMVIII_base		= 272;
	optional float SKPMVIII_add		= 273;
	optional float SKPMVIII_ratio		= 274;
	optional float SKPMVIII_sratio		= 275;
	optional float SKPMIX		= 276;
	optional float SKPMIX_base		= 277;
	optional float SKPMIX_add		= 278;
	optional float SKPMIX_ratio		= 279;
	optional float SKPMIX_sratio		= 280;
	optional float SKPMX		= 281;
	optional float SKPMX_base		= 282;
	optional float SKPMX_add		= 283;
	optional float SKPMX_ratio		= 284;
	optional float SKPMX_sratio		= 285;
	optional float RDSH		= 286;
	optional float RDSH_base		= 287;
	optional float RDSH_add		= 288;
	optional float RDSH_ratio		= 289;
	optional float RDSH_sratio		= 290;
	optional float RDKX		= 291;
	optional float RDKX_base		= 292;
	optional float RDKX_add		= 293;
	optional float RDKX_ratio		= 294;
	optional float RDKX_sratio		= 295;
	optional float NXCT		= 296;
	optional float NXCT_base		= 297;
	optional float NXCT_add		= 298;
	optional float NXCT_ratio		= 299;
	optional float NXCT_sratio		= 300;
	optional float NXKX		= 301;
	optional float NXKX_base		= 302;
	optional float NXKX_add		= 303;
	optional float NXKX_ratio		= 304;
	optional float NXKX_sratio		= 305;
	optional float MS		= 306;
	optional float MS_base		= 307;
	optional float MS_add		= 308;
	optional float MS_ratio		= 309;
	optional float MS_sratio		= 310;
	optional float SZGJ		= 311;
	optional float SZGJ_base		= 312;
	optional float SZGJ_add		= 313;
	optional float SZGJ_ratio		= 314;
	optional float SZGJ_sratio		= 315;
	optional float SOM		= 316;
	optional float SOM_base		= 317;
	optional float SOM_add		= 318;
	optional float SOM_ratio		= 319;
	optional float SOM_sratio		= 320;
	optional float WPPMI		= 321;
	optional float WPPMI_base		= 322;
	optional float WPPMI_add		= 323;
	optional float WPPMI_ratio		= 324;
	optional float WPPMI_sratio		= 325;
	optional float WPPMII		= 326;
	optional float WPPMII_base		= 327;
	optional float WPPMII_add		= 328;
	optional float WPPMII_ratio		= 329;
	optional float WPPMII_sratio		= 330;
	optional float WPPMIII		= 331;
	optional float WPPMIII_base		= 332;
	optional float WPPMIII_add		= 333;
	optional float WPPMIII_ratio		= 334;
	optional float WPPMIII_sratio		= 335;
	optional float WPPMIV		= 336;
	optional float WPPMIV_base		= 337;
	optional float WPPMIV_add		= 338;
	optional float WPPMIV_ratio		= 339;
	optional float WPPMIV_sratio		= 340;
	optional float WPPMV		= 341;
	optional float WPPMV_base		= 342;
	optional float WPPMV_add		= 343;
	optional float WPPMV_ratio		= 344;
	optional float WPPMV_sratio		= 345;
	// GENERATE END
}

//查看协助怪盗属性
message SCReqAssistInfo {
//    option (msgid) = 1925;
	optional int64 humanId 				= 1; // 玩家ID
	optional DUnitProp assistThiefAttr	= 2; // 协助怪盗属性
}